src/lib/dnssd/ActiveResolveAttempts.h

```diff
@@ -181,6 +186,28 @@ class ActiveResolveAttempts
         bool IsIpResolve() const { return resolveData.Is<IpResolve>(); }
         void Clear() { resolveData = DataType(); }
 
+        void WillCoalesceWith(const ScheduledAttempt & existing)
+        {
+            if (IsResolve() && existing.Matches(*this))
+            {
+                resolveData.Get<Resolve>().consumerCount = existing.resolveData.Get<Resolve>().consumerCount + 1;
+            }
+        }
+
+        void ConsumerRemoved()
+        {
+            if (!IsResolve())
+            {
+                return;
+            }
+
+            auto & count = resolveData.Get<Resolve>().consumerCount;
+            if (count > 0)
+            {
+                --count;
+            }
```

Comment:
would the else branch warrant a log (is it an internal error to remove a consumer when count is already thought to be 0?)

Reply:
I would comment here and maybe log if we try to decrease a 0 count (unless it really happens easily). I expect a pairing where we say 'add ref' and 'dec ref' to not really generally go negative.